test/strings/basic.jl

```diff
@@ -113,7 +113,10 @@ end
     @test endswith(z, z)
 end
 
-@test filter(x -> x ∈ ['f', 'o'], "foobar") == "foo"
+@testset "filter specialization on String issue #32460" begin
+     @test filter(x -> x ∈ ['f', 'o'], "foobar") == "foo"
```

Comment:
Perhaps take the opportunity to add a few "harder" tests with the string containing a few Greek letters for example.

Reply:
@KristofferC I think that one should be ready to go!